cryptography/src/secp256r1.rs

```diff
@@ -0,0 +1,672 @@
+//! secp256r1 implementation of the `Scheme` trait.
+
+use crate::{PrivateKey, PublicKey, Scheme, Signature};
+use commonware_utils::union_unique;
+use p256::{
+    ecdsa::{
+        signature::{Signer, Verifier},
+        SigningKey, VerifyingKey,
+    },
+    elliptic_curve::scalar::IsHigh,
+};
+use rand::{CryptoRng, Rng, SeedableRng};
+use std::borrow::Cow;
+
+const PUBLIC_KEY_LENGTH: usize = 33;
+const SIGNATURE_LENGTH: usize = 64;
+
+#[derive(Clone)]
+pub struct Secp256r1 {
+    signing_key: SigningKey,
+    verifying_key: VerifyingKey,
+}
+
+impl Scheme for Secp256r1 {
+    fn new<R: CryptoRng + Rng>(r: &mut R) -> Self {
+        let signing_key = SigningKey::random(r);
+        let verifying_key = signing_key.verifying_key().to_owned();
+        Self {
+            signing_key,
+            verifying_key,
+        }
+    }
+
+    fn from(private_key: PrivateKey) -> Option<Self> {
+        let signing_key = match SigningKey::from_slice(private_key.as_ref()) {
```

Comment:
I like the pattern we used in `ed25519` and `bls12381` where we explicitly check the length before passing to the inner library: https://github.com/commonwarexyz/monorepo/blob/73cb9c86897f2a686b6474f9bdd2f7712002ace8/cryptography/src/ed25519.rs#L44-L78

Reply:
Do you want to check the length of Public Key also ? Then restrict the sec1-encoded length to 33 bytes or accept both 33 and 65  bytes ?
Restricting the length to 33 only will change the tests. I will have to compress the provided x and y coordinates in a 33 bytes length so `invalid` public key could become valid if the invalid part was the y coordinate.

Reply:
I think we should just accept compressed length (33 bytes)?

Reply:
I agree, I see no reason to use 65 bytes length Public Keys.
I will change the tests regarding this restriction 👍